src/integer/mat_poly_over_z/evaluate.rs

```diff
@@ -5,12 +5,11 @@
 use super::MatPolyOverZ;
 use crate::{
     integer::{MatZ, Z},
-    macros::for_others::{implement_for_others, implement_for_owned},
     traits::{Evaluate, GetNumColumns, GetNumRows},
 };
 use flint_sys::fmpz_poly_mat::fmpz_poly_mat_evaluate_fmpz;
 
-impl Evaluate<&Z, MatZ> for MatPolyOverZ {
+impl<Integer: Into<Z>> Evaluate<Integer, MatZ> for MatPolyOverZ {
     /// Evaluates a [`MatPolyOverZ`] on a given input of [`Z`] entrywise. Note that the
     /// [`Z`] in this case is only a reference.
     ///
```

Comment:
Update documentation that it works on  [Into<Z>]. The comment with Z is only a reference is outdated now.

I would suggest to also update the doc comment to once evaluate directly on `3` and once on `Z::from(3)`.